Runtime/Async/Coroutine/Pooled YieldInstruction/YieldPool.cs

```diff
@@ -0,0 +1,70 @@
+using System.Runtime.CompilerServices;
+using System;
+using System.Collections.Generic;
+using StansAssets.Foundation.Patterns;
+
+namespace StansAssets.Foundation.Async
+{
+    public static class YieldPool 
+    {
+        private static Dictionary<Type, DefaultPool<PooledYieldInstruction>> m_instructions;
```

Comment:
```suggestion
        private static Dictionary<Type, ObjectPool<PooledYieldInstruction>> m_instructions;
```
Please replace DefaultPool with ObjectPool.